prac_08/convert_miles_km.py

```diff
@@ -0,0 +1,39 @@
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.properties import StringProperty
+from kivy.core.window import Window
+
+CONVERSION_RATE = 1.609
+
+
+class MilesToKm(App):
+    """App that converts miles to km."""
+    message = StringProperty()
+
+    def build(self):
+        """Build the kivy app from the kv file."""
+        Window.size = (300, 200)
+        self.title = "Convert Miles to Kilometers"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        return self.root
+
+    def handle_calculate(self, miles):
+        """Covert the mile into km."""
+        self.message = str(self.convert_to_float(miles) * CONVERSION_RATE)
+
+    def handle_increment(self, value):
+        """ handle the increment"""
+        miles = self.convert_to_float(self.root.ids.user_input.text) + value
```

Comment:
Did you notice you seem to be repeating yourself with this code `self.root.ids.user_input.text`? You either get it from the kivy file (`self.text`) or in the main program. Did you consider using some form of get_valid_number where it gets the input from 'self.root.ids.user_input.text' once and validates it? Or maybe you could add that in your `convert_to_float` function somehow? This does not mean what you have done is wrong just something to consider.